keps/sig-api-machinery/4222-cbor-serializer/README.md

```diff
@@ -752,6 +753,42 @@ We expect no non-infra related flakes in the last month as a GA graduation crite
 
 - request and response content-type negotiation with 1.17 sample API server
 
+### Custom JSON Marshalers
+
+If a type implements json.Marshaler or json.Unmarshaler without corresponding CBOR behaviors,
+serializing values of that type to and from CBOR using default behaviors risks mangling the data.
+
+As an example, consider the structure of a marshalled
+[IntOrString](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/intstr#IntOrString) with the custom
+behavior versus the default behavior:
+
+| Go                                       | Custom  | Default                               |
+|------------------------------------------|---------|---------------------------------------|
+| IntOrString{Type: Int, IntVal: 7}        | 7       | {"IntVal":7,"StrVal":"","Type:":0}    |
+| IntOrString{Type: String, StrVal: "foo"} | "foo"   | {"IntVal":0,"StrVal":"foo","Type:":1} |
+| IntOrString{Type: -1}                    | <error> | {"IntVal":0,"StrVal":"","Type:":-1}   |
+
+Imagine a similar type is declared out-of-tree. From this type, a CRD and typed client are
+generated. This typed client is used in a program to write to a custom resource, using JSON to
+encode the request body. On the server side, the request body is decoded into an Unstructured
+object, and within that object, the IntOrString value is represented by either a `string` or an
+`int64`.
+
+Now imagine that the same request is repeated, but with CBOR as the negotiated content type of the
+request body. On the server side, the IntOrString value is represented within the decoded
+Unstructured as a `map[string]interface{}` with three keys, `"IntVal"`, `"StrVal`", and `"Type"`. A
+Change in the request encoding resulted in a change to the object the client intended to send.
+
+The CBOR serializer must not use the default behaviors to marshal and unmarshal values that
```

Comment:
| The CBOR serializer must not use the default behaviors to marshal and unmarshal values

Great! Yes, this was the same conclusion I came to.  I didn't follow the above CRD example, but for intOrString, as it exists today for in-tree (and CRD) types, I don't see any way to get around the need to support custom marshal/unmarshal.  And it also seems valuable for aggregated API servers and such that we support the existing custom marshallers (json.Marshaller...)